snap/pack/pack.go

```diff
@@ -107,27 +107,33 @@ func CheckSkeleton(w io.Writer, sourceDir string) error {
 }
 
 func loadAndValidate(sourceDir string) (*snap.Info, error) {
-	// ensure we have valid content
+	// Parsing of snap info is duplicated in ReadInfoFromSnapFile. It is done
+	// here to retrieve the snap instance name, if available, to use in case of
+	// ReadInfoFromSnapFile error
 	yaml, err := ioutil.ReadFile(filepath.Join(sourceDir, "meta", "snap.yaml"))
 	if err != nil {
 		return nil, err
 	}
-
 	info, err := snap.InfoFromSnapYaml(yaml)
 	if err != nil {
 		return nil, err
 	}
+	instanceName := info.InstanceName()
```

Comment:
I wonder if we should introduce a small helper in internal.SnapNameFromYaml that does this by just trying to parse only:

```
struct {
   Name string `json:"name"`
}
```

and doing minimal checks on the name, possibly shared with snap.Validate via some other internal helper

Reply:
I attempted this route, but later formed the opinion that it was not worth keeping the ReadInfoFromSnapFile based approach, which means this is not a problem anymore.